### ISSUE TITLE
I cannot get this to work

### ISSUE THREAD

**Issue by ghost** [nucleartux/i18next-scanner-typescript#7]
Hiya,

Love the project and can't wait using this. However, the instructions are a bit hard to digest, as copy pasting the configuration example (and my custom config) will result in an errorless output, but nothing will have been parsed. I'm using CRA with very little modifications and have a single `.tsx` component with a `useTranslation` hook that is either not detected, or nothing is happening.

1) Do I mention tsx / ts anywhere in the config?
2) Do you have a full sample by any chance for a CRA-setup?

Thanks

**Comment by Poky85** [comment]
Generally it works.

But because it compiles Typescript to JS, proper functionality of scanner is very dependent on your TS compiler options. I imported my tsconfig.ts and now it works. In case of CRA tsconfig is "hidden" I guess. Most important options:

```javascript
transform: typescriptTransform({
    extensions: ['.ts', '.tsx'],
    tsOptions: {
      "target": "esnext",
      "jsx": "preserve", // <-- needed for <Trans> component scanner to work
    },
```

Also upgrade `i18next-scanner` package to the latest version to get better Javascript parser.

**Comment by ghost** [comment]
I found the issue. By reading your readme where it says:
```ts
  options: {
    func: {
      // don't pass ts or tsx here! <<<<<<<
      extensions: ['.js', '.jsx'],
    },
    trans: {
      // don't pass ts or tsx here!<<<<<
      extensions: ['.js', '.jsx'],
    },
```

I just left .js and .jsx in the complete 'original' configuration of i18next-scanner, which made it skip these files for some reason. My fully working config is:
```js
const typescriptTransform = require('i18next-scanner-typescript');

module.exports = {
  input: [
    'src/**/*.{js,jsx,ts,tsx}',
    // Use ! to filter out files or directories
    '!src/**/*.spec.{js,jsx,ts,tsx}',
    '!src/i18n/**',
    '!**/node_modules/**',
  ],
  output: './',
  options: {
    debug: true,
    func: {
      list: ['i18next.t', 'i18n.t', 't'],
      extensions: ['.js', '.jsx'],
    },
    trans: false,
    lngs: ['en', 'nl', 'fr', 'ar', 'es', 'pt', 'ru'],
    ns: ['translation'],
    defaultLng: 'en',
    defaultNs: 'translation',
    defaultValue: function (lng, ns, key) {
      return '[' + lng + ']' + key;
    },
    resource: {
      loadPath: 'src/i18n/{{lng}}/{{ns}}.json',
      savePath: 'src/i18n/{{lng}}/{{ns}}.json',
      jsonIndent: 2,
      lineEnding: '\n',
    },
    nsSeparator: false, // namespace separator
    keySeparator: false, // key separator
    interpolation: {
      prefix: '{{',
      suffix: '}}',
    },
  },
  transform: typescriptTransform({
    extensions: ['.ts', '.tsx'],
    tsOptions: {
      target: 'es5',
      module: 'esnext',
    },
  }),
};
```
